### PR TITLE
3694 loss function as cumulative metric

### LINKED ISSUE #3694: Generalize Metric to any loss function
**Is your feature request related to a problem? Please describe.**
If I introduce a new loss function, I have to change the metric as well. For example: `monai.losses.Dice` and `monai.metrics.DiceMetric`. Most of the losses have not a metric associated.

**Describe the solution you'd like**
I wonder if the metrics can be made general to any loss function. Where the loss function would be an argument of the metric. Other metric arguments would be the reduction method and similar things.

**Describe alternatives you've considered**
The alternative is to develop my own Metric every time I change the loss function. But not ideal imo.

### PR DESCRIPTION
**Pull request by wyli**
Signed-off-by: Wenqi Li <wenqil@nvidia.com>

Fixes #3694

### Description
adds a wrapper for loss function:
```py
dice_loss = DiceLoss(include_background=True)
loss_metric = LossMetric(loss_fn=dice_loss)
```
so that `loss_metric` it can be used as a metric in:
https://github.com/Project-MONAI/MONAI/blob/b030839e98e6a00c1ab0e53545f60b62dc4da4a4/monai/handlers/ignite_metric.py#L32

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by Nic-Ma** [comment]
Hi @wyli ,

Users can use the ignite metric directly:
https://github.com/pytorch/ignite/blob/master/ignite/metrics/loss.py
Just like how we use the `Accuracy` of ignite.
CC @vfdev-5 .

Thanks.

**Comment by wyli** [comment]
@Nic-Ma yes I'm aware of that, the assumptions on loss_fn is stronger there: -- requiring both y_pred and y, and -- the output must be a single value and --must be used with an ignite engine... 

In our case, previously I thought the cumulative average is good enough, https://github.com/Project-MONAI/MONAI/blob/89684d677f6c93c6fbd023a614c07485c5223557/monai/metrics/cumulative_average.py#L21
But it doesn't seem to work with IgniteMetric handler.

**Review by myron — CHANGES_REQUESTED** [review]
> assumptions on loss_fn is stronger there: -- requiring both y_pred and y, and -- the output must be a single value and --must be used with an ignite engine...

@wyli  thanks for the PR , but I don't think this is what people are asking for in the issue https://github.com/Project-MONAI/MONAI/issues/3694, they did not ask for Ignite support. (unless I misunderstand).  

And currently we can currently use 

```
ac = CumulativeAverage()
loss_value = someloss(x,y)
ac.append(ac)
```

this will work even if loss_value is an array/list/tensor (and removes nans).  I'll clarify what people in the issue are actually asking, unless this PR addresses something else

**Comment by wyli** [comment]
Hi @myron, as I mentioned in the description of this PR, it's about using the loss value within `monai.handlers.IgniteMetric`. `CumulativeAverage` is not compatible within that class. If you are not sure about the PR, we can discuss, but please don't block it using your admin permission.

**Comment by wyli** [comment]
Hi @Nic-Ma  do you have any further comments here?

**Review by Nic-Ma — APPROVED** [review]

File: monai/metrics/loss_as_metric.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
+# Copyright (c) MONAI Consortium
```
Comment (Nic-Ma) [inline monai/metrics/loss_as_metric.py:1]: As the metric name is `LossMetric`, I would suggest to change the file name to `loss_metric.py`.
What do you think?

Thanks.

**Comment by Nic-Ma** [comment]
> @Nic-Ma yes I'm aware of that, the assumptions on loss_fn is stronger there: -- requiring both y_pred and y, and -- the output must be a single value and --must be used with an ignite engine...
> 
> In our case, previously I thought the cumulative average is good enough,
> 
> https://github.com/Project-MONAI/MONAI/blob/89684d677f6c93c6fbd023a614c07485c5223557/monai/metrics/cumulative_average.py#L21
> 
> 
> But it doesn't seem to work with IgniteMetric handler.

OK, that makes sense.

Thanks.

**Review by myron — APPROVED** [review]

**Comment by myron** [comment]
> Hi @myron, as I mentioned in the description of this PR, it's about using the loss value within `monai.handlers.IgniteMetric`. `CumulativeAverage` is not compatible within that class. If you are not sure about the PR, we can discuss, but please don't block it using your admin permission.

thank you for the reply. In the PR description on the top , it says that it fixes https://github.com/Project-MONAI/MONAI/issues/3694
In my understanding  it does not fix that issue, since people were asking for something else.  Hence, I left the review and requested the change.  Do you prefer me not leaving reviews at all or not requesting changes?  ps: I changed the review status to approve.
